Engine/Helper.cs

```diff
@@ -637,7 +637,7 @@ public int Compare(Tuple<int, int> t1, Tuple<int, int> t2)
     /// <summary>
     /// Class used to do variable analysis on the whole script
     /// </summary>
-    public class ScriptAnalysis: ICustomAstVisitor2
+    public class ScriptAnalysis: ICustomAstVisitor
```

Comment:
I've run it on the applicable rules and it was okay. However, we do need to update a lot of other test cases. Going to do that in another pull request